Physics/Dynamics/ProductionFormFactorDecorator.cpp

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) 2013, 2017 The ComPWA Team.
+// This file is part of the ComPWA framework, check
+// https://github.com/ComPWA/ComPWA/license.txt for details.
+
+#include <cmath>
+#include <iterator>
+#include <numeric>
+#include <vector>
+
+#include "Coupling.hpp"
+#include "Physics/HelicityFormalism/HelicityKinematics.hpp"
+#include "ProductionFormFactorDecorator.hpp"
+
+namespace ComPWA {
+namespace Physics {
+namespace Dynamics {
+
+ProductionFormFactorDecorator::ProductionFormFactorDecorator(std::string name,
```

Comment:
I would rename the class to just `FormFactorDecorator`, since you can use this class not only in the production.